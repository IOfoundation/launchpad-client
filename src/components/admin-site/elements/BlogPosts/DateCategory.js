import React from 'react';
import {PropTypes} from 'prop-types';
import {withStyles} from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';

import SelectElement from '@Shared/FormElements/Select';

export const DateCategory = props => {
  const {errors, handleBlur, handleChange, values, classes, categories} = props;

  const selectCategories = categories.map(categorie => ({
    value: String(categorie.id),
    name: categorie.name,
  }));
  return (
    <div className="m-bot-8" style={{padding: 8}}>
      <Grid container={true} spacing={16} alignItems="center">
        <Grid item={true} xs={12} md={8} className={classes.date}>
          {'Started, November 11, 2018'}
        </Grid>
        <Grid item={true} xs={12} md={4}>
          <SelectElement
            errors={errors}
            handleBlur={handleBlur}
            handleChange={handleChange}
            id={'category'}
            label={'Category Title'}
            name={'category'}
            value={values.category}
            selectOptions={selectCategories}
          />
        </Grid>
      </Grid>
    </div>
  );
};

const styles = () => {
  return {
    date: {
      color: '#7B7C7E',
    },
  };
};

DateCategory.propTypes = {
  categories: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      name: PropTypes.string,
    })
  ),
  classes: PropTypes.shape({}),
  errors: PropTypes.shape({}),
  handleBlur: PropTypes.func,
  handleChange: PropTypes.func,
  touched: PropTypes.shape({}),
  values: PropTypes.shape({}),
};

export default withStyles(styles)(DateCategory);
