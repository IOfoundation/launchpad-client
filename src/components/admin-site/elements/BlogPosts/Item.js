import React, {PureComponent} from 'react';
import {PropTypes} from 'prop-types';
import {withStyles} from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';
import Select from '@material-ui/core/Select';
import MenuItem from '@material-ui/core/MenuItem';

class Item extends PureComponent {
  state = {
    option: '',
    open: false,
  };

  handleClose = () => {
    this.setState({open: false});
  };

  handleOpen = () => {
    this.setState({open: true});
  };

  updateValue = event => {
    const info = event.target;

    this.setState({[event.target.name]: info.value});
  };

  render() {
    const {title, description, label, date, category, classes} = this.props;

    return (
      <div className={classes.container}>
        <h3 className={classes.title}>{title}</h3>
        <p className={classes.description}>{description}</p>
        <Grid
          container={true}
          className={classes.dateContainer}
          justify="flex-start"
          alignItems="center"
        >
          <Grid className={classes.label}>{label}</Grid>
          <Grid className={classes.date}>
            <span>{date}</span>
          </Grid>
          <div className={classes.hideSelect}>
            <i
              className={`material-icons ${classes.floating} ${classes.front}`}
              onClick={this.handleOpen}
            >
              {'keyboard_arrow_down'}
            </i>
            <Select
              open={this.state.open}
              value={this.state.option}
              onChange={this.updateValue}
              onClose={this.handleClose}
              onOpen={this.handleOpen}
              inputProps={{
                name: 'service',
                id: 'service-options',
                classes: {
                  icon: classes.icon,
                },
              }}
              className={`${classes.floating} ${classes.back}`}
            >
              <MenuItem value="Edit">{'Edit'}</MenuItem>
              <MenuItem value="Delete">{'Delete'}</MenuItem>
            </Select>
          </div>

          <Grid className={classes.category}>{category}</Grid>
        </Grid>
      </div>
    );
  }
}

const styles = theme => {
  return {
    container: {
      borderBottom: '1px solid black',
      color: 'black',
      marginBottom: '24px',
      paddingBottom: '24px',
    },
    title: {
      fontFamily: theme.fonts.bold,
      fontSize: '16px',
      lineHeight: '24px',
      marginBottom: '8px',
    },
    description: {
      color: '#7B7C7E',
      fontSize: '14px',
      lineHeight: '20px',
      marginBottom: '8px',
    },
    dateContainer: {
      fontSize: '12px',
      lineHeight: '16px',
    },
    label: {
      fontFamily: theme.fonts.bold,
      marginRight: '4px',
    },
    category: {
      fontFamily: theme.fonts.bold,
    },
    date: {
      marginRight: '4px',
    },
    hideSelect: {
      height: '24px',
      marginRight: '16px',
      position: 'relative',
      width: '24px',
    },
    floating: {
      position: 'absolute',
      top: 0,
      left: 0,
    },
    front: {
      cursor: 'pointer',
      zIndex: '2',
      '&:hover': {
        color: '#00a472',
      },
    },
    back: {
      visility: 'hidden',
      opacity: '0',
      zIndex: '1',
    },
  };
};

Item.propTypes = {
  category: PropTypes.string,
  classes: PropTypes.shape({
    category: PropTypes.string,
    container: PropTypes.string,
    date: PropTypes.string,
    dateContainer: PropTypes.string,
    description: PropTypes.string,
    label: PropTypes.string,
    title: PropTypes.string,
  }),
  date: PropTypes.string,
  description: PropTypes.string,
  label: PropTypes.string,
  title: PropTypes.string,
};

export default withStyles(styles)(Item);
