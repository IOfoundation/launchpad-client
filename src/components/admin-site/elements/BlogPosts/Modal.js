import React from 'react';
import {PropTypes} from 'prop-types';
import {withStyles} from '@material-ui/core/styles';
import ModalUi from '@material-ui/core/Modal';
import Grid from '@material-ui/core/Grid';

const Modal = props => {
  const {classes, modalClosed, open, cancelClicked, deleteClicked} = props;

  return (
    <ModalUi open={open} onClose={modalClosed}>
      <div className={classes.modal}>
        <h2 className={classes.title}>
          {'Are you sure you want to delete this post?'}
        </h2>
        <Grid container={true} justify={'flex-end'}>
          <Grid item={true}>
            <button
              className={`btn btn__outline p-x-24 ${classes.btn}`}
              onClick={deleteClicked}
            >
              {'Yes, Delete It.'}
            </button>
          </Grid>
          <Grid item={true}>
            <button
              className={`btn btn__black p-x-24 ${classes.btn}`}
              onClick={cancelClicked}
            >
              {'Cancel'}
            </button>
          </Grid>
        </Grid>
      </div>
    </ModalUi>
  );
};

const styles = theme => ({
  modal: {
    backgroundColor: theme.palette.background.paper,
    borderRadius: '5px',
    boxShadow: theme.shadows[5],
    left: '50%',
    maxHeight: '70vh',
    outline: 'none',
    overflow: 'auto',
    padding: '24px 28px',
    position: 'absolute',
    top: '45%',
    transform: 'translate(-50%, -50%)',
    width: '300px',
  },
  btn: {
    padding: '8px 12px',
  },
  title: {
    fontSize: '24px',
    lineHeight: '36px',
    marginBottom: '16px',
  },
});

Modal.propTypes = {
  cancelClicked: PropTypes.func,
  classes: PropTypes.shape({
    btn: PropTypes.string,
    modal: PropTypes.string,
  }),
  deleteClicked: PropTypes.func,
  modalClosed: PropTypes.func,
  open: PropTypes.bool,
};

export default withStyles(styles)(Modal);
