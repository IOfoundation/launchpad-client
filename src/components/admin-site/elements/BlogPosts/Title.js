import React from 'react';
import {PropTypes} from 'prop-types';
import Grid from '@material-ui/core/Grid';

import TitleField from '@Shared/FormElements/TitleField';

export const Title = props => {
  const {errors, touched, handleBlur, handleChange, values} = props;

  return (
    <div className="m-bot-8">
      <Grid container={true}>
        <Grid item={true} xs={12}>
          <TitleField
            autocomplete="off"
            error={touched.title && Boolean(errors.title)}
            errors={errors}
            handleBlur={handleBlur}
            handleChange={handleChange}
            id="title"
            name="title"
            label="Add a title..."
            value={values.title}
            multiline={true}
          />
        </Grid>
      </Grid>
    </div>
  );
};

Title.propTypes = {
  classes: PropTypes.shape({}),
  errors: PropTypes.shape({}),
  handleBlur: PropTypes.func,
  handleChange: PropTypes.func,
  touched: PropTypes.shape({}),
  values: PropTypes.shape({}),
};

export default Title;
