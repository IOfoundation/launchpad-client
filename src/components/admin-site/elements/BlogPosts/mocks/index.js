export {drafts} from './drafts';
export {posted} from './posted';
