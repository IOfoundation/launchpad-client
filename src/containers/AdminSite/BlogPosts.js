import React from 'react';
import Layout from './Layout';
import BlogPostsFormContainer from '../../components/admin-site/elements/BlogPostsFormContainer';

const BlogPosts = () => {
  return (
    <Layout>
      <BlogPostsFormContainer />
    </Layout>
  );
};

export default BlogPosts;
