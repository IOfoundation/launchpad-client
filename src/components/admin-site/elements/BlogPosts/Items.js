import React from 'react';
import {PropTypes} from 'prop-types';

import Item from './Item';

const Items = props => {
  const {items} = props;

  return (
    <div>
      {items.map(item => (
        <Item
          key={item.id}
          category={item.category}
          date={item.date}
          description={item.description}
          label={item.label}
          title={item.title}
        />
      ))}
    </div>
  );
};

Items.propTypes = {
  items: PropTypes.arrayOf(
    PropTypes.shape({
      category: PropTypes.string,
      date: PropTypes.string,
      description: PropTypes.string,
      label: PropTypes.string,
      id: PropTypes.number,
      title: PropTypes.string,
    })
  ),
};

export default Items;
