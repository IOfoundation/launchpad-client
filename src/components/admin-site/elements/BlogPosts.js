import React, {PureComponent} from 'react';
import {PropTypes} from 'prop-types';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';

import LandingComponent from '../Landing';
import Title from '../Title';
import Items from './BlogPosts/Items';
import Pagination from '../../businesses/Pagination';
import CustomTabs from '@Shared/Tabs';
import Loading from '@Shared/Loading';

import * as getAdminPost from '@Actions/admin-blogs';
import {htmlStripper, truncate, getDate} from '@Utils';

class BlogPosts extends PureComponent {
  componentDidMount() {
    this.props.actions.getAdminPost(1);
  }

  menuChanged = index => {
    this.props.actions.getAdminPost(1, this._tabOptions[index]);
  };

  _tabOptions = ['Drafts', 'Posted'];

  render() {
    const handleChangePage = () => {};
    const {drafts, posted, noResults} = this.props;
    let draftsElements = <Loading />;
    let postedElements = <Loading />;
    let pagination = null;

    if (noResults) {
      draftsElements = (
        <p className="text-regular paragraph">{'No posts available.'}</p>
      );
      postedElements = (
        <p className="text-regular paragraph">{'No posts available.'}</p>
      );
    } else {
      if (drafts.data.length > 0) {
        draftsElements = <Items items={drafts.data} />;
      }

      if (posted.data.length > 0) {
        postedElements = <Items items={posted.data} />;
      }

      pagination = (
        <Pagination
          appliedFilters={{category: 'admin-posts', page: 1}}
          handleChangePage={handleChangePage}
          metadata={{
            pagination: {
              last: {
                page: 10,
              },
              currentPage: 1,
            },
            totalOrganization: '10',
          }}
          noMargin={true}
        />
      );
    }

    return (
      <LandingComponent navigation={true}>
        <Title
          titleText="Your Blog Posts"
          hideCancelAction={true}
          submitLabel="Create Blog Post"
        />
        <CustomTabs tabs={this._tabOptions} changed={this.menuChanged}>
          {draftsElements}
          {postedElements}
        </CustomTabs>
        {pagination}
      </LandingComponent>
    );
  }
}

function postToBlogPosts(posts, draft = false) {
  return posts.map(post => {
    const category =
      (post.categories && post.categories[0].name) || 'No Category';
    const date = getDate(post.posted_at);
    let description = htmlStripper(post.body);
    let title = post.title;

    if (description.split('').length > 190) {
      description = truncate(description, 190);
    }

    if (title.split('').length > 130) {
      title = truncate(title, 130);
    }

    return {
      id: post.id,
      title,
      description,
      label: draft ? 'Last Edited:' : 'Posted:',
      date: `${date.monthLarge} ${date.dayNumber}, ${date.year}`,
      category,
    };
  });
}

const mapStateToProps = _state => {
  const drafts = _state.adminBlogs.drafts;
  const posted = _state.adminBlogs.posted;

  return {
    drafts: {
      data: postToBlogPosts(drafts.data, true),
      noResults: drafts.data.length > 0,
      page: drafts.page,
      totalPages: drafts.totalPages,
    },
    posted: {
      data: postToBlogPosts(posted.data),
      noResults: posted.data.length > 0,
      page: posted.page,
      totalPages: posted.totalPages,
    },
    noResults: _state.adminBlogs.noResults,
  };
};

const mapDispatchToProps = _dispatch => {
  return {
    actions: bindActionCreators(getAdminPost, _dispatch),
  };
};

BlogPosts.propTypes = {
  actions: PropTypes.shape({
    getAdminPost: PropTypes.func,
  }),
  drafts: PropTypes.shape({
    data: PropTypes.arrayOf(PropTypes.shape({})),
    page: PropTypes.number,
    totalPages: PropTypes.number,
  }),
  noResults: PropTypes.bool,
  posted: PropTypes.shape({
    data: PropTypes.arrayOf(PropTypes.shape({})),
    page: PropTypes.number,
    totalPages: PropTypes.number,
  }),
  router: PropTypes.shape({
    push: PropTypes.func,
  }),
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(BlogPosts);
