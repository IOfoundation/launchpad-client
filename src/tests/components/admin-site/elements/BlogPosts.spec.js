import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import MuiThemeProvider from '@material-ui/core/styles/MuiThemeProvider';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import {Provider} from 'react-redux';

import BlogPosts from '../../../../components/admin-site/elements/BlogPosts';

import {muiTheme} from '../../../helpers/theme';

const props = {
  location: {
    pathname: '/admin/profile',
  },
};
const mockStore = configureStore([thunk]);
let store;

describe('Admin-site <BlogPosts />', () => {
  it('renders snapshot of BlogPosts', () => {
    store = mockStore({});
    const wrapper = shallow(
      <MuiThemeProvider theme={muiTheme}>
        <Provider store={store}>
          <BlogPosts {...props} />
        </Provider>
      </MuiThemeProvider>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
