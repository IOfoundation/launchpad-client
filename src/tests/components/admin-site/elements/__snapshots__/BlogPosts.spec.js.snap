// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin-site <BlogPosts /> renders snapshot of BlogPosts 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BlogPosts
    location={
      Object {
        "pathname": "/admin/profile",
      }
    }
  />
</Provider>
`;
